client/wallet/wallet.go

```diff
@@ -181,59 +187,68 @@ func (wallet *Wallet) ConsolidateFunds(options ...consolidateoptions.Consolidate
+		// build inputs from consumed outputs
+		inputs := wallet.buildInputs(consumedOutputs)
+		// aggregate all the funds we consume from inputs
+		totalConsumedFunds := consumedOutputs.TotalFundsInOutputs()
+		toAddress := wallet.chooseToAddress(consumedOutputs, address.AddressEmpty) // no optional toAddress from options
 
-	outputs := ledgerstate.NewOutputs(ledgerstate.NewSigLockedColoredOutput(ledgerstate.NewColoredBalances(totalConsumedFunds), toAddress.Address()))
+		outputs := ledgerstate.NewOutputs(ledgerstate.NewSigLockedColoredOutput(ledgerstate.NewColoredBalances(totalConsumedFunds), toAddress.Address()))
 
-	// determine pledgeIDs
-	aPledgeID, cPledgeID, err := wallet.derivePledgeIDs(consolidateOptions.AccessManaPledgeID, consolidateOptions.ConsensusManaPledgeID)
-	if err != nil {
-		return
-	}
+		// determine pledgeIDs
+		aPledgeID, cPledgeID, pErr := wallet.derivePledgeIDs(consolidateOptions.AccessManaPledgeID, consolidateOptions.ConsensusManaPledgeID)
+		if pErr != nil {
+			err = pErr
+			return
+		}
 
-	txEssence := ledgerstate.NewTransactionEssence(0, time.Now(), aPledgeID, cPledgeID, inputs, outputs)
-	outputsByID := consumedOutputs.OutputsByID()
+		txEssence := ledgerstate.NewTransactionEssence(0, time.Now(), aPledgeID, cPledgeID, inputs, outputs)
+		outputsByID := consumedOutputs.OutputsByID()
 
-	unlockBlocks, inputsAsOutputsInOrder := wallet.buildUnlockBlocks(inputs, outputsByID, txEssence)
+		unlockBlocks, inputsAsOutputsInOrder := wallet.buildUnlockBlocks(inputs, outputsByID, txEssence)
 
-	tx = ledgerstate.NewTransaction(txEssence, unlockBlocks)
+		tx = ledgerstate.NewTransaction(txEssence, unlockBlocks)
 
-	// check syntactical validity by marshaling an unmarshaling
-	tx, _, err = ledgerstate.TransactionFromBytes(tx.Bytes())
-	if err != nil {
-		return nil, err
-	}
+		// check syntactical validity by marshaling an unmarshaling
+		tx, _, err = ledgerstate.TransactionFromBytes(tx.Bytes())
+		if err != nil {
+			return nil, err
+		}
 
-	// check tx validity (balances, unlock blocks)
-	ok, err := checkBalancesAndUnlocks(inputsAsOutputsInOrder, tx)
-	if err != nil {
-		return nil, err
-	}
-	if !ok {
-		return nil, errors.Errorf("created transaction is invalid: %s", tx.String())
-	}
+		// check tx validity (balances, unlock blocks)
+		ok, err := checkBalancesAndUnlocks(inputsAsOutputsInOrder, tx)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			return nil, errors.Errorf("created transaction is invalid: %s", tx.String())
+		}
 
-	wallet.markOutputsAndAddressesSpent(consumedOutputs)
-	err = wallet.connector.SendTransaction(tx)
-	if err != nil {
-		return nil, err
-	}
-	if consolidateOptions.WaitForConfirmation {
-		err = wallet.WaitForTxConfirmation(tx.ID())
+		wallet.markOutputsAndAddressesSpent(consumedOutputs)
+		err = wallet.connector.SendTransaction(tx)
+		if err != nil {
+			return nil, err
+		}
+		if consolidateOptions.WaitForConfirmation {
+			err = wallet.WaitForTxConfirmation(tx.ID())
```

Comment:
```suggestion
			err = wallet.WaitForTxConfirmation(tx.ID())
			if err != nil {
				return nil, err
			}
```

Reply:
Double check that this is what should happen in case of error.

Reply:
I've refactored it.